File: tests/journal.lib/chronicler_detail.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2023 all rights reserved


// get the journal
#include <pyre/journal.h>
// support
#include <cassert>


// alias
using chronicler_t = pyre::journal::chronicler_t;


// verify that the detail level set in the environment matches what the test harness gave me
// on the command line
int main(int argc, char *argv[]) {
    // make sure we get one command line argument
    assert (argc == 2);
    // get the detail
    auto detail = chronicler_t::detail();
    // interpret the command line argument as the intended level
    pyre::journal::detail_t expected = std::atoi(argv[1]);
    // verify they match
    assert (detail == expected);

    // all done
    return 0;
}


// end of file

File: tests/journal.api/chronicler_detail.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2023 all rights reserved


def test():
    """
    Verify that we can control the global detail threshold
    """
    # access
    import journal

    # get the global state
    chronicler = journal.chronicler

    # verify that the detail level is at its default value
    assert chronicler.detail == 1
    # set it to some other value
    chronicler.detail = 5
    # verify the assignment sticks
    assert chronicler.detail == 5

    # all done
    return


# main
if __name__ == "__main__":
    # run the test
    test()


# end of file
